msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -554,51 +555,128 @@ public void run() {
-     * @param authority             The authority of the sought IAccount.
-     * @return The IAccount stored in the cache or null, if no such matching entry exists.
+     * @param oid The object id of the sought IAccount.
+     * @param callback AccountLoadedCallback
      */
-    @Nullable
-    public IAccount getAccount(@NonNull final String homeAccountIdentifier,
-                               @Nullable final String authority) {
+    public void getAccount(@NonNull final String oid,
+                           @NonNull final AccountsLoadedCallback<IAccount> callback) {
         final String methodName = ":getAccount";
 
         ApiDispatcher.initializeDiagnosticContext();
 
-        String realm = StringUtil.getTenantInfo(homeAccountIdentifier).second;
-
-        Authority authorityObj = Authority.getAuthorityFromAuthorityUrl(authority);
+        if (StringUtil.isEmpty(oid)) {
+            throw new IllegalArgumentException("Illegal oid to get the sought account.");
+        }
 
-        if (authorityObj instanceof AzureActiveDirectoryAuthority) {
-            final AzureActiveDirectoryAuthority aadAuthority = (AzureActiveDirectoryAuthority) authorityObj;
-            final AzureActiveDirectoryAudience audience = aadAuthority.getAudience();
-            realm = audience.getTenantId();
+        if (MSALControllerFactory.brokerEligible(
+                mPublicClientConfiguration.getAppContext(),
+                mPublicClientConfiguration.getDefaultAuthority(),
+                mPublicClientConfiguration)) {
+            com.microsoft.identity.common.internal.logging.Logger.info(
+                    TAG + methodName,
+                    "Eligible to use broker. Try to get the account from local and broker cache."
+            );
+            getBrokerAndLocalAccount(oid, callback);
         } else {
-            com.microsoft.identity.common.internal.logging.Logger.warn(
+            // load local account and find the matching account
+            com.microsoft.identity.common.internal.logging.Logger.info(
                     TAG + methodName,
-                    "Provided authority was not AAD - defaulting to parsed home_account_id"
+                    "Not eligible to use broker. Try to get the account from local cache."
             );
+            getLocalAccount(oid, callback);
         }
+    }
 
-        AccountRecord accountToReturn = null;
+    private void getBrokerAndLocalAccount(@NonNull final String oid,
+                                          @NonNull final AccountsLoadedCallback callback) {
+        final String methodName = ":getBrokerAndLocalAccount";
 
-        if (null != realm) {
-            accountToReturn = AccountAdapter.getAccountInternal(
-                    mPublicClientConfiguration.getClientId(),
-                    mPublicClientConfiguration.getOAuth2TokenCache(),
-                    homeAccountIdentifier,
-                    realm
-            );
+        final Handler handler;
```

Comment:
`getAccounts()` and `getBrokerLocalAccounts()` have the same piece of logic - let's split this into a convenience function like `getActiveLooper()` or `getPreferredLooper()`